leech/src/modules/service_detection/probe_files/telnet.probe

```diff
@@ -0,0 +1,9 @@
+service: telnet
+prevalence: often
+probes:
+  - protocol: TCP
+    # pattern should be 0xff (IAC) <DO | DON'T | WILL | WON'T > < COMMAND >, repeat
+    # may end in `" `
+    #
+    # DO (0xfd), DON'T (0xfe), WILL (0xfb), WON'T (0xfc)
+    regex: [(?-u:\xff)[(?-u:\xfd|\xfe|\xfb|\xfc)].]*|[Tt][Ee][Ll][Nn][Ee][Tt]
```

Comment:
This regex is way too broad. It would for example match:
`HTTP/1.0 200 \r\n telnet`
which should obviously not be detected as telnet

Without knowing telnet's protocol I would suggest at least anchoring the regex to the start, but please research that guess first.

Reply:
Or add subregexes to fine-tune a match